### DISCUSSION TITLE
Is it safe to call Rollback() after call Commit() ?

### DISCUSSION THREAD

**Discussion by stasvo** [go-gorm/gorm#5683] [Q&A] [answered]
Is it safe to call Rollback() after call Commit() or call Rollback() multople times? 
I need it for call Rollback in defer.
I use  gorm with postgresql, and in documentation of pgx there information that its safe:
"defer tx.Rollback() is safe even if tx.Commit() will be called first in a non-error condition. "
But in documentation of gorm I don't see information about it.

**Comment by jinzhu** [comment, marked as answer]
Yes, it's safe, but won't rollback anything.

**Reply by stasvo** [reply]
This is exactly what I need. Possibility to call rollback in defer in both cases: after commit and if commit didn't happen.

**Reply by jinzhu** [reply]
@stasvo suggest to use the `Transaction` method if possible https://gorm.io/docs/transactions.html#Transaction

**Reply by stasvo** [reply]
@jinzhu And also I have one little additional question about rollback in case if control the transaction manually.
In case if commit successfully done and after it called tx.Rollback(), whether a rollback request will be sent to the server and as a result of which the server reports that rollback is impossible or the gorm keeps a state that the commit has already been made and in this case there will be no rollback request to the server?

**Reply by jinzhu** [reply]
will be sent to the server, gorm doesn't keeps any states.
